libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/IpvSessionService.java

```diff
@@ -50,51 +49,59 @@ public IpvSessionService(ConfigService configService) {
         this.sleeper = new Sleeper();
     }
 
-    public IpvSessionItem getIpvSession(String ipvSessionId) {
-        return dataStore.getItem(ipvSessionId);
+    public IpvSessionItem getIpvSession(String ipvSessionId) throws IpvSessionNotFoundException {
+        IpvSessionItem ipvSessionItem = dataStore.getItem(ipvSessionId);
+        if (ipvSessionItem == null) {
+            throw new IpvSessionNotFoundException(
+                    "The session not found in the database for the supplied session Id");
```

Comment:
Careful, this is now a breaking change if any clients were expecting to handle `null` being returned. For example in `ProcessJourneyEventHandler` there's some code that handles `null` (but also just throws an error).

Can you make sure that there's nowhere similar?

Reply:
only here we are checking for null, removed it as not needed as handled in exception.